projects/fal/src/fal/_serialization.py

```diff
@@ -110,72 +28,62 @@ def _pickle_model_field(
             "alias": field.alias,
             "field_info": field.field_info,
         }
-        pickler.save_reduce(_pydantic_make_field, (args,), obj=field)
 
-    @dill.register(pydantic.fields.ModelPrivateAttr)
-    def _pickle_model_private_attr(
-        pickler: dill.Pickler,
-        field: pydantic.fields.ModelPrivateAttr,
-    ) -> None:
+        def deser(kwargs):
+            return pydantic.fields.ModelField(**kwargs)
+
+        return deser, (args,)
+
+
+    def ModelPrivateAttrReducer(field: pydantic.fields.ModelPrivateAttr):
         args = {
             "default": field.default,
             "default_factory": field.default_factory,
         }
-        pickler.save_reduce(_pydantic_make_private_field, (args,), obj=field)
 
+        def deser(kwargs):
+            return pydantic.fields.ModelPrivateAttr(**kwargs)
 
-@mainify
-def patch_pydantic_class_attributes():
-    # Dill attempts to modify the __class__ of deserialized pydantic objects
-    # on this side but it meets with a rejection from pydantic's semantics since
-    # __class__ is not recognized as a proper dunder attribute.
-    try:
-        import pydantic.utils
-    except ImportError:
-        return
+        return deser, (args,)
 
-    pydantic.utils.DUNDER_ATTRIBUTES.add("__class__")
+    cloudpickle.Pickler.dispatch[pydantic.fields.ModelField] = ModelFieldReducer
+    cloudpickle.Pickler.dispatch[pydantic.fields.ModelPrivateAttr] = ModelPrivateAttrReducer
 
 
-@mainify
-def patch_exceptions():
-    # Adapting tblib.pickling_support.install for dill.
-    from types import TracebackType
+def patch_pickle() -> None:
```

Comment:
Would be nice to use some context manager or something instead of blasting `patch_pickle` at module level. We'll need to do something about it later (would be nice make it available to the users too so they could add their own handlers as well when needed).